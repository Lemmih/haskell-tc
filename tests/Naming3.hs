{-# LANGUAGE RankNTypes #-}
module Naming3 where

fn :: forall b. b -> (forall a. a -> a)
fn b a = a
