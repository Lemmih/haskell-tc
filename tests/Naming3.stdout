Bindings:
:: Î›b. Î»1::b. Î›a. ((âˆ€ b a. b â†’  a â†’  a) @ b a) 1
  [4mfn b a = a[0m
:: b
  fn [4mb[0m a = a
:: a
  fn b [4ma[0m = a

Proofs:
coercion: a
  fn b a = [4ma[0m
